### PR TITLE
Allow Bay Description to be Blank

### LINKED ISSUE #111: Description should be optional on Bay Creation
Also, if a space is inputted into description, it redirects to a json error page.

### PR DESCRIPTION
**Pull request by jaydenteo**
Fixes #111 

## Proposed Changes
  - Can add a space character to description successfully now. 
  - Resolves - "Also, if a space is inputted into description, it redirects to a json error page."
  - Could not figure out how to get past input validation ("Please fill in this field")!

### CODE REVIEW

**Comment by zollf** [comment]
so i've made everything required by default with our custom form tag:
> https://github.com/zollf/CITS3200/blob/main/app/admin/templatetags/admin.py
> Specifically: `<input type="text" id="{{field.name}}" name="{{field.name}}" value="{{data|key:field.name}}" required {% if static %}disabled{% endif %}/>`
> https://github.com/zollf/CITS3200/blob/main/app/resources/templates/form.html 

So it's just HTML and embedded validation. A possible solution is to add another parameter for required, or some way to pass whether a field is required or not. I would NOT recommend extending `fields` param to handle requirement e.g., 
`{% create_form 'carparks' carpark 'name:text:Username:required|desc:text:Description:required' "/admin/carpark/add" %}` as label is optional already and would make the code more unreadable.

A possible option is `{% create_form 'carparks' carpark 'name:r_text:Username|desc:r_text:Description' "/admin/carpark/add" %}` and then spliting `r_text` (and other field types) into `type` and `required` based on whether it has `r` or not.

Another option is just having another param `{% create_form 'carparks' carpark 'name:text:Username|desc:text:Description' "/admin/carpark/add" required=name|desc %}`. Tho you'll have to change all existing ones as they could error without param, whereas the other wont have that issue but would make everything not required.

Any better solutions are welcomed

File: app/parking/models.py [hunk 1]
```diff
@@ -16,7 +16,7 @@ class CarBay(models.Model):
         on_delete=models.CASCADE,
     )
     bay_number = models.CharField(max_length=50)
-    description = models.TextField()
+    description = models.TextField(blank=True)
```
Comment (APM246) [inline app/parking/models.py:19]: You will need to run ``` docker-compose run django python manage.py makemigrations ``` to create the migration file that updates the database schema. That file can be committed as well.

File: app/admin/templatetags/admin.py [hunk 2]
```diff
@@ -20,16 +20,20 @@ def create_form(title: str, data: dict, fields: str, **kwargs):
...
             })
         else:
             fields_matrix.append({
                 'name': field_arr[0],
-                'type': field_arr[1],
+                'type': field_arr[1] if "r_" not in field_arr[1] else field_arr[1][2:],
                 'label': field_arr[2] if len(field_arr) >= 3 else field_arr[0],
+                'required': True if "r_" not in field_arr[1] else False
             })
+    print(fields_matrix)
```
Comment (APM246) [inline app/admin/templatetags/admin.py]: This print call should be removed

File: app/admin/templatetags/admin.py [hunk 3]
```diff
@@ -20,16 +20,20 @@ def create_form(title: str, data: dict, fields: str, **kwargs):
     fields_matrix = []
     for field in fields:
         field_arr = field.split(':')
+
         if field_arr[1] == 'hidden':
             fields_matrix.append({
                 'name': field_arr[0],
-                'type': field_arr[1],
+                'type': field_arr[1] if "r_" not in field_arr[1] else field_arr[1][2:],
```
Comment (APM246) [inline app/admin/templatetags/admin.py]: Maybe a different letter than r_ should be used since it is denoting a *non*-required field. E.g. n_

**Review by APM246 — CHANGES_REQUESTED** [review]

**Review by APM246 — APPROVED** [review]

**Pull request merged**